dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -165,4 +191,44 @@ public CommunityResource wrapResource(CommunityRest community, String... rels) {
         return new CommunityResource(community, utils, rels);
     }
 
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected CommunityRest put(Context context, HttpServletRequest request, String apiCategory, String model, UUID id,
+                       JsonNode jsonNode)
+        throws RepositoryMethodNotImplementedException, SQLException, AuthorizeException {
+        CommunityRest communityRest = new Gson().fromJson(jsonNode.toString(), CommunityRest.class);
+        Community community = cs.find(context, id);
+        if (community == null) {
+            throw new ResourceNotFoundException(apiCategory + "." + model + " with id: " + id + " not found");
+        }
+        if (StringUtils.equals(id.toString(), communityRest.getId())) {
+            List<MetadataEntryRest> metadataEntryRestList = communityRest.getMetadata();
+            community = (Community) dspaceObjectUtils.replaceMetadataValues(context, community, metadataEntryRestList);
+        } else {
+            throw new IllegalArgumentException("The UUID in the Json and the UUID in the url do not match: "
+                                                   + id + ", "
+                                                   + communityRest.getId());
+        }
+        return converter.fromModel(community);
+    }
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected void delete(Context context, UUID id) throws AuthorizeException {
+        Community community = null;
+        try {
+            community = cs.find(context, id);
+            if (community == null) {
+                throw new ResourceNotFoundException(
+                    CommunityRest.CATEGORY + "." + CommunityRest.NAME + " with id: " + id + " not found");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
Again, improve error message

Reply:
Change to something like:
`throw new RuntimeException("Unable to find Community with id=" + id, e);`